rest-java/src/main/java/org/hiero/mirror/restjava/controller/HooksController.java

```diff
@@ -72,6 +98,47 @@ ResponseEntity<HooksResponse> getHooks(
         return ResponseEntity.ok(response);
     }
 
+    @GetMapping("/{hookId}/storage")
+    ResponseEntity<HooksStorageResponse> getHookStorage(
+            @PathVariable EntityIdParameter ownerId,
+            @PathVariable long hookId,
+            @RequestParam(name = KEY, required = false) @Size(max = MAX_REPEATED_QUERY_PARAMETERS)
+                    List<
+                                    @Pattern(
+                                            regexp = "^((eq|gt|gte|lt|lte):)?(0x)?[0-9a-fA-F]{1,64}$",
+                                            message = "Key must be in format <hex> or <op>:<64 or less char hex>")
+                                    String>
+                            keys,
+            @RequestParam(name = TIMESTAMP, required = false) @Size(max = MAX_REPEATED_QUERY_PARAMETERS)
```

Comment:
We only support a min and max or single timestamp. Use default "" and remove null checks for `timestamps`.

```suggestion
            @RequestParam(name = TIMESTAMP, required = false, default = "") @Size(max = 2)
```